desktop/src/app.rs

```diff
@@ -0,0 +1,488 @@
+use crate::cli::Opt;
+use crate::custom_event::RuffleEvent;
+use crate::gui::{GuiController, MovieView, MENU_HEIGHT};
+use crate::player::PlayerController;
+use crate::util::{
+    get_screen_size, parse_url, pick_file, winit_key_to_char, winit_to_ruffle_key_code,
+    winit_to_ruffle_text_control,
+};
+use anyhow::{Context, Error};
+use ruffle_core::{PlayerEvent, StageDisplayState};
+use ruffle_render::backend::ViewportDimensions;
+use ruffle_render_wgpu::backend::WgpuRenderBackend;
+use std::rc::Rc;
+use std::sync::{Arc, Mutex};
+use std::time::Instant;
+use winit::dpi::{LogicalSize, PhysicalPosition, PhysicalSize, Size};
+use winit::event::{ElementState, KeyboardInput, ModifiersState, VirtualKeyCode, WindowEvent};
+use winit::event_loop::{ControlFlow, EventLoop, EventLoopBuilder};
+use winit::window::{Fullscreen, Icon, Window, WindowBuilder};
+
+pub struct App {
+    opt: Opt,
+    window: Rc<Window>,
+    event_loop: Option<EventLoop<RuffleEvent>>,
+    gui: Arc<Mutex<GuiController>>,
+    player: PlayerController,
```

Comment:
nit: _Slightly_ confusing name. (Same below where this is constructed.) No big deal though...